frame/support/src/event.rs

```diff
@@ -796,4 +804,17 @@ mod tests {
 	fn outer_event_metadata() {
 		assert_eq!(EXPECTED_METADATA, TestRuntime::outer_event_metadata());
 	}
+
+	#[test]
+	fn test_codec() {
+		let runtime_1_event_module_2 = TestEvent::event_module2(
+			event_module2::Event::<TestRuntime>::TestEvent(3)
+		);
+		assert_eq!(runtime_1_event_module_2.encode()[0], 2);
+
+		let runtime_2_event_module_2 = TestEventSystemRenamed::event_module2(
+			event_module2::Event::<TestRuntime2>::TestEvent(3)
+		);
+		assert_eq!(runtime_2_event_module_2.encode()[0], 5);
```

Comment:
```suggestion
		let runtime_2_event_module_2 = TestEventSystemRenamed::event_module2(
			event_module2::Event::<TestRuntime2>::TestEvent(3)
		);
		assert_eq!(runtime_2_event_module_2.encode()[0], 5);
		
		let runtime_2_event_module_3 = TestEventSystemRenamed::event_module3(
			event_module3::Event::<TestRuntime2>::HiEvent
		);
		assert_eq!(runtime_2_event_module_3.encode()[0], 6);
```

Reply:
actually in impl_outer_event you give only the codec attribute. so the rules about incrementation from previous index doesn't hold here.
So the index is 3 as its position in the enum.

Reply:
Ahh okay, but in `construct_runtime!` we actually will set the indices always?

Reply:
indeed construct_runtime set all the indices explicitly